sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -638,6 +644,17 @@ private[hive] class HiveClientImpl(
             s"No partition is dropped. One partition spec '$s' does not exist in table '$table' " +
             s"database '$db'")
         }
+        // Check whether the partition we are going to drop is empty.
+        // We make a dummy one for the empty partition. See [SPARK-29786] for more details.
+        parts.foreach { partition =>
+          val partPath = partition.getPath.head
+          if (isExistPath(partPath)) {
+            val fs = partPath.getFileSystem(conf)
+            fs.mkdirs(partPath)
+            fs.deleteOnExit(partPath)
+          }
```

Comment:
I'm confused. When the partition exists (`isExistPath` returns true), why you need to mkdir it again?

Reply:
> I'm confused. When the partition exists (`isExistPath` returns true), why you need to mkdir it again?

Sorry, I mistakenly delete `!` when adjusting the code ..